### PR TITLE
Refactor of model and controller into separate file or article

### PR DESCRIPTION
**Pull request by kamakazee**
Get api/aarticles/:article_id?query=comment_count

### CODE REVIEW

**Review by corkesw — COMMENTED** [review]
Hi Cam, I'd like you to go back to the drawing board on this one. We are not handling the comment_count as a query. The ticket just asks us to add a comment_count to the articles endpoint. This will require a refactor to return this value as part of the response from this endpoint.

File: controllers/article-controller.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const {
+    selectArticles,
+    selectArticleById,
+    setArticleById,
+  } = require("../models/article-model.js");
+
+
+const getArticles = (request, response) => {
+  selectArticles().then(({ rows: articles }) => {
```
Comment (Rosem10) [inline controllers/article-controller.js:9]: Please keep your de-structuring consistent across your functions. I prefer the way you've done it for selectArticleById below (de-structured the article from rows in the model before returning back up to the controller.)

File: models/article-model.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          });
+        }
+      });
+    
+    
+  };
+  
+  const setArticleById = (inc_votes, article_id) => {
+
+    if (!inc_votes) {
```
Comment (Rosem10) [inline models/article-model.js:29]: This isn't necessarily a 400! The body could include no key, and have no effect on the votes, but still not break anything.

File: models/article-model.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  };
+  
+  const selectArticleById = (article_id) => {
+  
+      return db
+      .query(`SELECT articles.*, COUNT(articles.article_id) AS comment_count FROM articles JOIN comments ON comments.article_id = articles.article_id WHERE articles.article_id=$1 GROUP BY articles.article_id`, [article_id])
+      .then(({ rows: article }) => {
+       
+        if (article.length > 0) {
+          return article[0];
```
Comment (Rosem10) [inline models/article-model.js:15]: In this case, "article" doesn't reflect the article, it reflect the rows array that article is being returned within. Therefore article[0] isn't readable - what is 0 of article? This should either be rows[0] or article should be destructured (currently just being renamed from rows) on  line 12

**Pull request merged**